Bottom: 739a68b4f57f0f0eb2ed2dff067ebc5d7d38c39d
Top:    f5735a65fa21caff3bc2da3d25711351195dcadd
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-08-02 14:19:26 -0700

Refresh of setopt

---

diff --git a/src/rsocket.c b/src/rsocket.c
index e3406a4..3cc3a22 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -3309,18 +3309,22 @@ int rsetsockopt(int socket, int level, int optname,
 		switch (optname) {
 		case RDMA_SQSIZE:
 			rs->sq_size = min((*(uint32_t *) optval), RS_QP_MAX_SIZE);
+			ret = 0;
 			break;
 		case RDMA_RQSIZE:
 			rs->rq_size = min((*(uint32_t *) optval), RS_QP_MAX_SIZE);
+			ret = 0;
 			break;
 		case RDMA_INLINE:
 			rs->sq_inline = min(*(uint32_t *) optval, RS_QP_MAX_SIZE);
 			if (rs->sq_inline < RS_MIN_INLINE)
 				rs->sq_inline = RS_MIN_INLINE;
+			ret = 0;
 			break;
 		case RDMA_IOMAPSIZE:
 			rs->target_iomap_size = (uint16_t) rs_scale_to_value(
 				(uint8_t) rs_value_to_scale(*(int *) optval, 8), 8);
+			ret = 0;
 			break;
 		case RDMA_ROUTE:
 			if ((rs->optval = calloc(optlen, 1))) {
